language/move-prover/docgen/tests/testsuite.rs

```diff
@@ -24,11 +24,35 @@ const FLAGS: &[&str] = &[
 fn test_runner(path: &Path) -> datatest_stable::Result<()> {
     let mut args = vec!["mvp_test".to_string()];
     args.extend(FLAGS.iter().map(|s| (*s).to_string()).collect_vec());
-    args.push(path.to_string_lossy().to_string());
+
+    let is_root_template = path
+        .file_name()
+        .unwrap()
+        .to_string_lossy()
+        .to_string()
+        .ends_with("_template.md");
+    if !is_root_template {
+        args.push(path.to_string_lossy().to_string());
+    } else {
+        // add as sources all .notest_move files starting with the name of the template.
+        let base_name = path.file_stem().unwrap().to_string_lossy().to_string();
+        let dir_name = path.parent().unwrap();
+        dir_name.read_dir().unwrap().for_each(|res| {
+            let f = res.unwrap();
+            let name = f.file_name().to_string_lossy().to_string();
+            if name.starts_with(&base_name) && name.ends_with(".notest_move") {
```

Comment:
I think this can be simplified to: `if f.starts_with(path.with_extension("notest_move"))` (which should let you kill 38 and 42 also).

Reply:
This can match `<basename>*.notest_move`, so I would need a regular expression. Seemed to be overkill so I think I stay with this more brute force approach.